draft-packages/menu/KaizenDraft/Menu/components/MenuHeading.tsx

```diff
@@ -0,0 +1,32 @@
+import React, { useEffect, useState } from "react"
+import classnames from "classnames"
+import { v4 } from "uuid"
+import { Heading, HeadingProps } from "@kaizen/typography"
+import styles from "./MenuHeading.module.scss"
+
+export interface MenuHeadingProps extends Partial<HeadingProps> {
+  children: React.ReactNode
+  setMenuListAriaLabelledBy?: (headingId: string) => void
+}
+
+export const MenuHeading = (props: MenuHeadingProps): JSX.Element => {
+  const { id, setMenuListAriaLabelledBy, classNameOverride, ...restProps } =
+    props
+  const [headingId] = useState<string>(id || v4())
+
+  useEffect(() => {
+    setMenuListAriaLabelledBy && setMenuListAriaLabelledBy(headingId)
+  }, [])
```

Comment:
Yeah I think your hunch is right, in that it shouldn't accept a setter, apologies for the misguided comment about encapsulating the `id` in here, it makes more sense for the `MenuList` to decide the `id` and pass it down. 😅

Reply:
All good, it was worth a shot. I've removed that commit now and fixed the merge conflicts